mongodump/metadata_dump.go

```diff
@@ -19,10 +19,10 @@ import (
 
 // Metadata holds information about a collection's options and indexes.
 type Metadata struct {
-	Options        bson.M   `json:"options,omitempty"`
-	Indexes        []bson.D `json:"indexes"`
-	UUID           string   `json:"uuid,omitempty"`
-	CollectionName string   `json:"collectionName"`
+	Options        bson.M   `bson:"options,omitempty"`
+	Indexes        []bson.D `bson:"indexes"`
+	UUID           string   `bson:"uuid,omitempty"`
+	CollectionName string   `bson:"collectionName"`
```

Comment:
Funnily enough I don't believe the struct tags weren't doing anything before. The [2nd point in the BSON doc here](https://godoc.org/go.mongodb.org/mongo-driver/bson#hdr-Structs) suggests that the tags start with `bson: ...`

Reply:
Yep that was a good idea. Seems like bson/json struct tags don't completely align in terms of the tags available, among some other use cases. The `MarshallExtJSON()` func is in the bson package anyway, so I suppose that's counter-counterintuitive enough for me :)